src/output/mod.rs

```diff
@@ -209,11 +209,20 @@ fn pubsub_stats(snapshot: &mut MetricsSnapshot) {
 }
 
 pub fn json(config: Config, ratelimit: Option<&Ratelimiter>) {
-    if config.general().json_output().is_none() {
+    if config.general().output_file().is_none() {
         return;
     }
 
-    let file = std::fs::File::create(config.general().json_output().unwrap());
+    let output_file = config.general().output_file().clone().unwrap();
+    let (json_file, pq_file) = match output_file.ends_with(".parquet") {
```

Comment:
I think I'd prefer explicit output format in the config instead of using the extension to autodetect intent.

Reply:
I was going back and forth between this and don't have a strong preference. The reason I picked this was that I wasn't sure what semantics we should have if the output file is specified, but the format is not (or vice versa)? Ignore, pick a default, or fail-stop?

I'm leaning towards one of the latter since running a test and then not having any output is bad...

Reply:
I think defaulting to JSON would be okay.